src/routes/oauth/config.ts

```diff
@@ -54,12 +55,22 @@ export const PROVIDERS_CONFIG: Record<
     profile: ({ jwt, profile }) => {
       const payload = jwt?.id_token?.payload;
 
-      const isProfileError = typeof profile === 'object' && profile.error;
-      const userProfile = !isProfileError ? JSON.parse(profile) : null;
-      const displayName =
-        isProfileError && !userProfile
-          ? payload.email
-          : `${userProfile.name.firstName} ${userProfile.name.lastName}`;
+      let displayName;
+
+      if (profile) {
+        try {
+          const userProfile = JSON.parse(profile);
+
+          displayName = userProfile.name
+            ? `${userProfile.name.firstName} ${userProfile.name.lastName}`
+            : displayName;
+        } catch (error) {
+          logger.warn(error);
```

Comment:
isn't it unnecessary to always log a warning since existing apple users will always return an empty `profile` based on apple docs?

Reply:
I think it's good we log things like this as people may not be aware of this but we should probably add context, logging the error without saying what we were doing might not be very helpful, i.e.:

```
logger.warn(`problem trying to parse user data from apple's reponse: ${error}`);
```

Reply:
Yeah, that makes sense

Reply:
Yeah that's true, will adjust

Reply:
[5b17b5a](https://github.com/nhost/hasura-auth/pull/425/commits/5b17b5ae65daa383755f2a5de718f10578a0699c)